packages/lesswrong/components/tagging/PostsTagsList.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import groupBy from 'lodash/groupBy';
+import { tagStyle } from './FooterTag';
+import sortBy from 'lodash/sortBy';
+import classNames from 'classnames';
+import filter from 'lodash/filter';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    flexWrap: "wrap",
+    alignItems: "center",
+    marginTop: 4,
+    marginBottom: 4
+  },
+  tagFilter: {
+    ...tagStyle(theme),
+    backgroundColor: theme.palette.background.pageActiveAreaBackground,
+    marginTop: 2,
+    marginBottom: 2,
+    // paddingLeft: 10,
+    // paddingRight: 8
+  },
+  count: {
+    ...theme.typography.body2,
+    fontSize: "1rem",
+    marginLeft: 6,
+    color: theme.palette.grey[500]
+  },
+  selected: {
+    backgroundColor: theme.palette.grey[700],
+    color: theme.palette.background.pageActiveAreaBackground
+  },
+  button: {
+    ...theme.typography.body2,
+    color: theme.palette.grey[500],
+    fontSize: "1rem",
+    marginLeft: 6,
+  }
+});
+
+export const PostsTagsList = ({classes, posts, currentFilter, handleFilter}:{
+  classes: ClassesType,
+  posts: PostsList[]|null,
+  currentFilter: string|null,
+  handleFilter: (string) => void
+}) => {
+  const { LWTooltip } = Components
+
+
+  const tags = posts?.flatMap(post => post.tags)
+  const groupedTags = groupBy(tags, (tag) => tag._id)
+  const tagsWithCount = Object.entries(groupedTags).map(tag => ({
+    ...tag[1][0],
+    count: tag[1].length
+  }))
```

Comment:
I'm finding this quite hard to read, could you change it to something like this? (not completely correct because you're spreading the tag whereas I haven't, but the main thing I'm finding confusing is the method of grouping)
```suggestion
  const allTags = posts?.flatMap(post => post.tags) ?? []
  const uniqueTags = [...new Set(allTags)]
  const tagsWithCount: {tag: TagPreviewFragment, count: number}[] = uniqueTags.map(tag => ({
    tag,
    count: allTags.filter(t => t._id === tag._id).length
  }))
```

Reply:
Updated